sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1414,10 +1414,20 @@ class GpuStringSplitMeta(
     extends StringSplitRegExpMeta[StringSplit](expr, conf, parent, rule) {
   import GpuOverrides._
 
-  private var delimInfo: Option[(String, Boolean)] = None
+  private var pattern = ""
+  private var isRegExp = false
 
   override def tagExprForGpu(): Unit = {
-    delimInfo = checkRegExp(expr.regex)
+    checkRegExp(expr.regex) match {
+      case Some((p, isRe)) => pattern = p; isRegExp = isRe
```

Comment:
nit: I'm not used to seeing semicolons in Scala and I think the code would be easier to read without that

```suggestion
      case Some((p, isRe)) =>
        pattern = p
        isRegExp = isRe
```